### PR TITLE
[connectivity] Add connectivity_tizen package

### PR DESCRIPTION
**Pull request by bwikbs**
Initial release.

### CODE REVIEW

File: packages/connectivity/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+flutter:
+  plugin:
+    platforms:
+      tizen:
+        pluginClass: ConnectivityTizenPlugin
+        fileName: connectivity_tizen_plugin.h
+dependencies:
+  connectivity: ^0.4.9
+  connectivity_platform_interface: ^1.0.3
```
Comment (swift-kim) [inline packages/connectivity/pubspec.yaml:14]: The latest connectivity plugin version is 2.0.2 and the platform interface version is 1.0.6. Would it be better to update the dependencies before connectivity_tizen is published on pub.dev for the first time?

File: packages/connectivity/lib/connectivity_tizen.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// You have generated a new plugin project without
```
Comment (swift-kim) [inline packages/connectivity/lib/connectivity_tizen.dart:1]: I guess this file was added accidentally.

**Review by swift-kim — COMMENTED** [review]
You may also add these lines to https://github.com/flutter-tizen/plugins/blob/master/README.md (after testing the plugin on watch/watch emulator/TV/TV emulator):

- Under "List of packages" (insert after "battery_tizen")
  ```sh
  | [**connectivity_tizen**](packages/connectivity) | [connectivity](https://github.com/flutter/plugins/tree/master/packages/connectivity) (1st-party) | [![pub package](https://img.shields.io/pub/v/connectivity_tizen.svg)](https://pub.dev/packages/connectivity_tizen) | No |
  ```

- Under "Device limitations"
  ```
  | [**connectivity_tizen**](packages/connectivity) | ✔️ | ✔️ | ✔️ | ✔️ |
  ```


Looks good otherwise!

File: packages/connectivity/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import 'package:connectivity/connectivity.dart';
+```
+
+For detailed usage, see https://github.com/flutter/plugins/tree/master/packages/connectivity/connectivity#usage.
+
+## Supported devices
+
+This plugin is supported on these types of devices:
+
+- Galaxy Watch (running Tizen 4.0 or later)
```
Comment (swift-kim) [inline packages/connectivity/README.md:27]: Please specify if any privilege is required to use this plugin in README. There's an example in the checklist or https://github.com/flutter-tizen/plugins/tree/master/packages/path_provider#required-privileges.

File: packages/connectivity/pubspec.yaml (hunk 1 above)
Comment (bwikbs) [reply packages/connectivity/pubspec.yaml]: You are right. It was updated with latest version.

File: packages/connectivity/lib/connectivity_tizen.dart (hunk 2 above)
Comment (bwikbs) [reply packages/connectivity/lib/connectivity_tizen.dart]: Thanks.

File: packages/connectivity/README.md (hunk 3 above)
Comment (bwikbs) [reply packages/connectivity/README.md:26]: Done.

**Comment by bwikbs** [comment]
> You may also add these lines to https://github.com/flutter-tizen/plugins/blob/master/README.md (after testing the plugin on watch/watch emulator/TV/TV emulator):
> 
> * Under "List of packages" (insert after "battery_tizen")
>   ```shell
>   | [**connectivity_tizen**](packages/connectivity) | [connectivity](https://github.com/flutter/plugins/tree/master/packages/connectivity) (1st-party) | [pub package](https://pub.dev/packages/connectivity_tizen) | No |
>   ```
> * Under "Device limitations"
>   ```
>   | [**connectivity_tizen**](packages/connectivity) | ✔️ | ✔️ | ✔️ | ✔️ |
>   ```
> 
> Looks good otherwise!

My TV environment has some problem, so I couldn't check at TV. I'll find a way and it will be updated soon.

**Comment by swift-kim** [comment]
The example app worked as expected on TV emulator, but on mobile and wearable emulators, the status always showed "wifi" (even when they were in flight mode). Maybe it's a defect in the system library or SDK.

**Comment by bwikbs** [comment]
> The example app worked as expected on TV emulator, but on mobile and wearable emulators, the status always showed "wifi" (even when they were in flight mode). Maybe it's a defect in the system library or SDK.

It's pretty OK at TM1, TW3. I think it's a bug of emulator.

**Comment by swift-kim** [comment]
Then we may mark this plugin as ❌ or ⚠️ in the device limitations section ("Watch emulator" column) in README.md. You can revert your changes to README.md if we are going to update it later at once.

File: packages/connectivity/tizen/src/connectivity_tizen_plugin.cc [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    {
+      connection_destroy(m_connection);
+      m_connection = nullptr;
+    }
+  }
+
+  void registerObsever(std::unique_ptr<flutter::EventSink<flutter::EncodableValue>> &&events)
+  {
+    ensureConnectionHandle();
+    if (connection_set_type_changed_cb(m_connection, connetionTypeChangedCB, this) != CONNECTION_ERROR_NONE)
```
Comment (seungsoo47) [inline packages/connectivity/tizen/src/connectivity_tizen_plugin.cc:75]: I am not sure but did you set clang_format_style as Google?

Comment (bwikbs) [reply packages/connectivity/tizen/src/connectivity_tizen_plugin.cc]: done.

**Review by swift-kim — APPROVED** [review]

**Pull request merged**